packages/arb-token-bridge-ui/src/components/TransferPanel/CustomDestinationAddressConfirmationDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from 'react'
+
+import { Dialog, UseDialogProps } from '../common/Dialog'
+import { ExternalLink } from '../common/ExternalLink'
+import { useNetworks } from '../../hooks/useNetworks'
+import { getNetworkName } from '../../util/networks'
+import { shortenAddress } from '../../util/CommonUtils'
+import { useArbQueryParams } from '../../hooks/useArbQueryParams'
+import { Checkbox } from '../common/Checkbox'
+
+export function CustomDestinationAddressConfirmationDialog(
+  props: UseDialogProps
+) {
+  const [{ destinationAddress = '' }] = useArbQueryParams()
+
+  const [networks] = useNetworks()
+
+  const networkName = getNetworkName(networks.destinationChain.id)
+
+  const [checkboxChecked, setCheckboxChecked] = useState(false)
+
+  function closeWithReset(confirmed: boolean) {
+    props.onClose(confirmed)
+    setCheckboxChecked(false)
+  }
+
+  return (
+    <Dialog
+      {...props}
+      title="Confirm Destination Address"
+      className="flex flex-col gap-4"
+      onClose={closeWithReset}
+      actionButtonProps={{
+        disabled: !checkboxChecked
+      }}
+    >
+      <div className="mb-4">
+        <p className="pb-2">
+          You are attempting to deposit funds to the same address{' '}
+          {shortenAddress(destinationAddress)} on {networkName}.
```

Comment:
Would be nice to have a link to the address on the explorer.

Reply:
Updated